renderer/src/GLRenderer.cpp

```diff
@@ -0,0 +1,303 @@
+#include <renderer/GLRenderer.hpp>
+
+#include <mutex>
+
+#include <core/gl_extensions.hpp>
+
+#include <data_types/rendering/BufferObject.hpp>
+
+#include <boost/log/trivial.hpp>
+
+namespace rcbe::rendering {
+class GLRendererImplementation {
+public:
+    GLRendererImplementation(RendererConfig &&config, const core::WindowPtr& window );
+    ~GLRendererImplementation();
+
+    void start();
+    void stop();
+
+    [[nodiscard]] const RendererConfig& get_config() const;
+
+    void add_object(rcbe::geometry::Mesh &&object);
+private:
+
+    void render_loop();
+
+    void init_gl();
+
+    void reshape_window();
+
+    void init_lights();
+
+    void set_perspective(const double fov, const double aspect, const double near, const double far);
+
+    void draw_buffers(const VertexBufferObject& vbo, const IndexBufferObject& ibo);
+
+    core::WindowPtr window_;
+    RendererConfig config_;
+
+    std::mutex control_mutex_;
+    bool running_ = false;
+    bool vbo_supported_ = false;
+
+    std::vector<rcbe::geometry::Mesh> meshes_;
+
+    std::vector<VertexBufferObject> vertex_buffers_;
+    std::vector<IndexBufferObject> index_buffer_;
+
+    double zoom_ = -50;
```

Comment:
Since user input handling is not implemented yet, and wee need to view scene and see that there is something there, this value has to stay for time being, I will add a TODO